runtime/src/alloc.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const MAX_MEMORY: usize = 0x400000;
```

Comment:
This value cannot be hardcoded in, as we allow the user to set it larger dynamically through a macro or the SDK.

Reply:
I updated `alloc.rs`, I realize there is a simpler solution: Grab the `sp` and perform GAP between heap and stack check. Thus, the MAX_MEMROY is no longer needed. 
Updated in 0944f29 .